statsd/uds_async.go

```diff
@@ -0,0 +1,119 @@
+package statsd
+
+import (
+	"fmt"
+	"net"
+	"time"
+)
+
+// asyncUdsWriter is an internal class wrapping around management of UDS connection
+type asyncUdsWriter struct {
+	// Address to send metrics to, needed to allow reconnection on error
+	addr net.Addr
```

Comment:
I would probably put `addr`, `conn` and `writeTimeout` in a separate structure that both the async and blocking writer structs would include in their definitions. 

edit: see note regarding using the async writer as a blocking writer.

Reply:
Let's discuss https://github.com/DataDog/datadog-go/pull/81/files#r266391555 and come back to this if it still make sense